### PR TITLE
Scriptable object IList

### PR DESCRIPTION
**Pull request by mutmedia**
Made ScriptableObjectList implement IList interface #33 

Not sure how to handle the unity auto generated/changed files.

### CODE REVIEW

**Comment by AdamRamberg** [comment]
Thanks for the PR! 

Had a couple of comments for some minor stuff. Please fix and I will merge your PR ⭐️

**Comment by mutmedia** [comment]
I cant find your comments, any chance you forgot to submit them to the PR?

File: Source/Base/ScriptableObjectList.cs [hunk 1]
```diff
@@ -90,5 +91,32 @@ public List<T> List
...
+
+        IEnumerator IEnumerable.GetEnumerator() => list.GetEnumerator();
+
+        public IEnumerator<T> GetEnumerator() => list.GetEnumerator();
+
+        public int IndexOf(T item) => list.IndexOf(item);
+
+        public void RemoveAt(int index)
+        {
+            var item = list[index];
```
[the remark this answers is not in the document]
Comment (mutmedia) [reply Source/Base/ScriptableObjectList.cs]: Doesnt list[index] already throw that same exception in that case? Does it make sense to repeat that check? (might be a very dumb question)

[the remark this answers is not in the document]
Comment (AdamRamberg) [reply Source/Base/ScriptableObjectList.cs]: You are correct, you get the same exception (my bad). We'll just leave it as it is.

**Pull request merged**